xarray/backends/api.py

```diff
@@ -777,6 +777,9 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
 
     combined._file_obj = _MultiFileCloser(file_objs)
     combined.attrs = datasets[0].attrs
+    for v in combined.variables:
+        if v in datasets[0]:
+            combined[v].encoding = datasets[0][v].encoding
```

Comment:
Are we sure this is a good idea in general? What if data is legitimately written with a different encoding?

It also looks a bit unrelated to the rest of this change.

Reply:
#2436 is about mfdataset losing all encoding information. So this is a band-aid that makes the arbitrary choice of using the first file (but the same choice as is made for attrs).

> "What if data is legitimately written with a different encoding?"

If encoding changes between different files read with mfdataset, then I guess we should drop it in that case. It seems like this problem should be solved at the merge/concat level than up here.

What do you think? Remove this change and merge?

Reply:
It's also a small part of #2921 (current behaviour is called surprising) that's why this change is in this PR.

Reply:
I also think it makes more sense to handle this at the merge/concat level rather than here. So I would lean towards removing this from this change (and then merging it).